crates/block-producer/src/server.rs

```diff
@@ -211,8 +212,27 @@ impl BlockProducerRpcServer {
     }
 
     async fn serve(self, listener: TcpListener) -> Result<(), tonic::transport::Error> {
+        // Configure the trace layer with callbacks.
+        let trace_layer = TraceLayer::new_for_grpc()
+            .make_span_with(miden_node_utils::tracing::grpc::block_producer_trace_fn)
```

Comment:
We can consider moving this entire layer into the `utils` crate since its likely going to be identical.

Reply:
Potentially the way to do it:
https://stackoverflow.com/questions/71178212/how-to-configure-tower-http-tracelayer-in-a-separate-function

Reply:
Oh damn I see. Little corner cases everywhere.

Thanks, yeah I think lets go with the stackoverflow answer - and just document on the function why we're doing it this way.